alea/test/stream_serializer.cpp

```diff
@@ -142,6 +144,7 @@ class twogauss_serialize_case
 
         archive >> out; // deserialize
 
+        std::cerr << alps::alea::PRINT_VERBOSE << "\nout\n" << out << "\n";
         EXPECT_EQ(in, out);
```

Comment:
(Quoting https://github.com/ALPSCore/ALPSCore/pull/477#commitcomment-27816874):
> K: Just in case, why didn't you do the same for `batch_result`?
> M: Because `batch_result` does not normalize any results, i.e., the comparison works for `count() == 0`

Sorry for harping on this again :), but how do we know it?
It would be nice to have a test verifying that `in==out` is true iff `in` and `out` are equal, for usual cases and some reasonable corner cases like `count()==0` .

Reply:
Let's make this a seperate issue though, shall we (#482)? Otherwise this PR will drag on forever ...